src/pages/rasa.tsx

```diff
@@ -37,12 +46,36 @@ const Body = styled.div`
   overflow: hidden;
 `
 
-export const RasaPage: React.FC = () => {
-  const { t, i18n } = useTranslation()
+interface Metadata {
+  [key: string]: string
+}
+
+// TODO: change intent from env to remove / and pass it right here
+const createPayload = (intent: string, metadata: Metadata) => {
+  return `/${intent}{"${RASA_METADATA_ENTITY}":${JSON.stringify(metadata)}}`
+}
 
+const createDefaultPayload = (timezone?: string) => {
+  const metadata = {}
+  if (timezone) metadata[RASA_TIMEZONE_PARAMETER] = timezone
+  return createPayload(RASA_GREET_INTENT, metadata)
+}
+
+const createCheckInPayload = (userId: string) => {
+  const metadata = { [RASA_USER_ID_PARAMETER]: userId }
+  return createPayload(RASA_CHECKIN_INTENT, metadata)
+}
+
+export const RasaPage: React.FC<Props> = ({ timezone, userId }) => {
+  const { t, i18n } = useTranslation()
   const language = i18n.languages[0]
   const socketUrl = RASA_SOCKET_ENDPOINTS[language] || RASA_SOCKET_ENDPOINTS.en
 
+  // When userId is set, the user is coming back for check-in.
+  const initPayload = userId
```

Comment:
I think all the logic regarding how to build the `initPayload` should be in the router instead. I don't think it's the component job to do this.

Reply:
That makes a lot os sense. thanks!